src/jit/optimizer.cpp

```diff
@@ -3703,9 +3694,9 @@ void Compiler::optUnrollLoops()
                     ++loopRetCount;
                 }
 
-                /* Visit all the statements in the block */
+                // Visit all the statements in the loop.
 
-                for (GenTreeStmt* stmt = block->firstStmt(); stmt; stmt = stmt->gtNextStmt)
+                for (GenTreeStmt* stmt = loopStmt; stmt != nullptr; stmt = stmt->gtNextStmt)
```

Comment:
Can you explain why this is correct, and why we shouldn't continue to use `block->firstStmt()` here?

Reply:
I can't. Good catch, thank you.

 Looks like `loopStmt` (or `loop` before this change) is a dead variable. Our loop includes all blocks from head to bottom and that is what we are trying to sum here.

I will fix it shorty.